networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/NodeBasedStableSubnetSubscriber.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.eth2.gossip.subnets;
+
+import static java.util.Collections.emptySet;
+
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NavigableSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.units.bigints.UInt256;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.validator.SubnetSubscription;
+
+public class NodeBasedStableSubnetSubscriber implements StableSubnetSubscriber {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final AttestationTopicSubscriber persistentSubnetSubscriber;
+  private final NavigableSet<SubnetSubscription> subnetSubscriptions =
+      new TreeSet<>(
+          Comparator.comparing(SubnetSubscription::getUnsubscriptionSlot)
+              .thenComparing(SubnetSubscription::getSubnetId));
```

Comment:
what about explicitly implementing `Comparable<SubnetSubscription>` in `SubnetSubscription`?

Reply:
Done. That's better!